src/foam/dao/AuthenticatedSink.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.nanos.auth.AuthService;
+
+import javax.security.auth.AuthPermission;
+import java.security.Permission;
+
+public class AuthenticatedSink
+    extends ProxySink
+{
+  protected String name_;
+  protected String method_;
+
+  public AuthenticatedSink(String name, String method, Sink delegate) {
+    this.name_ = name;
```

Comment:
Should take X as arg and set, otherwise how will it find the AuthService?

Reply:
Done